File: Weijia/1137. N-th Tribonacci Number.py
class Solution(object):
    def tribonacci(self, n):
        if n<2:
            return n
        elif n==2:
            return 1
        m=[0,1,1]
        for i in range(3,n+1):
            m=[m[-2],m[-1],m[-3]+m[-2]+m[-1]]
        return m[-1]

File: solution_zzj/1137._N-th_Tribonacci_Number.cc
class Solution {
public:
    int tribonacci(int n) {
        unordered_map<int, int> memo;
        memo[0] = 0;
        memo[1] = 1;
        memo[2] = 1;
        for (int i=3; i<=n; i++) {
            memo[i] = memo[i-3] + memo[i-2] + memo[i-1];
        }
        return memo[n];
    }
};
